textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/field/Span.java

```diff
@@ -3,22 +3,40 @@
 public class Span {
     //The name of the field (in the tuple) where this span is present
     private String fieldName;
-    //The start of the span
+    //The start of the span. It is the position of the first character of span in the document.
     private int start;
-    //The end of the span
+    //The end of the span.It is the position of the first character of span in the document
     private int end;
     //The key we are searching for eg: regex
     private String key;
     //The value matching the key
     private String value;
-    
-    
-    public Span(String fieldName, int start, int end, String key, String value) {
+    // The token position of the span
+    private int tokenOffset;
+
+    /*
+    Example:
+        Value = "The quick brown fox jumps over the lazy dog"
+        Now the Span for brown should be
+        start = 10 : position of character 'b'
+        end = 15 : position of character 'n'

```

Comment:
@akshaybetala ... We should represent the gaps between the characters to specify a span (as seen from above example by @chenlica ). It is same as representing the range from first character index to last character index + 1.

For example to represent `york` in `new york city`, we should follow the following representation.
`(indexOf(y), indexOf(k)+ 1)` => (4, 8)

Which will be same as gap representation as demonstrated below
```
  0   1   2   3   4  5  6  7  8  9  10  11  12  13
    n   e   w       y  o  r  k     c   i   t   y
```